include/deal.II/multigrid/mg_coarse.h

```diff
@@ -424,8 +374,23 @@ MGCoarseGridIterativeSolver<
   Assert(preconditioner != nullptr, ExcNotInitialized());
 
   dst = 0;
-  internal::MGCoarseGridIterativeSolver::solve(
-    *solver, *matrix, *preconditioner, dst, src);
+
+  if constexpr (std::is_same_v<VectorType, typename SolverType::vector_type>)
+    {
+      solver->solve(*matrix, dst, src, *preconditioner);
+    }
+  else
+    {
+      typename SolverType::vector_type src_;
+      typename SolverType::vector_type dst_;
```

Comment:
@kronbichler One could make these also class variables. What do you think?

Reply:
I don't think it's very important, because the solver will allocate quite some memory anyway. Unless you use a solver in batched form with the right `VectorMemory` class, I doubt one would see any performance impact. So I suggest we go with this code; we can always adapt if we figure out that this is a performance bottleneck later.